embassy-stm32/src/i2c/v2.rs

```diff
@@ -619,8 +627,8 @@ impl<'d, T: Instance, TXDMA, RXDMA> I2c<'d, T, TXDMA, RXDMA> {
         TXDMA: super::TxDma<T>,
         RXDMA: super::RxDma<T>,
     {
-        self.write_dma_internal(address, bytes, true, true).await?;
-        self.read_dma_internal(address, buffer, true).await?;
+        self.write(address, bytes).await?;
```

Comment:
I don't think this is correct. Old code does "write, restart, read", new code does "write, stop, start, read"

Reply:
I think the write part does the same, but I did miss a different parameter on the read call. I'll just copy over the if statement.

Reply:
Ah, the write won't do the same if it takes the empty path of course.

Reply:
Can you confirm if the old code was correct? Seems to me like it would do "write, stop, restart, read"

Reply:
Haven't personally tested it, it's supposed to work but it's possible it was already wrong, yes.

Reply:
I think the latest commit should have the same behaviour as the old code, at least. The calls in the empty paths in `write_read` now also match the calls in `blocking_write_read`.